### DISCUSSION TITLE
RaspAP doesn't restart at boot after powering up (Raspberry Pi 3B+ and PI Zero W 2)

### DISCUSSION THREAD

**Discussion by Fabri54** [RaspAP/raspap-webgui#1544] [Q&A] [answered]
Hi everybody, I started, mostly for curiosity and fun, to build a WiFi Extender or a Range Extender (call it the way you like it) based on a Pi 3B+ and on a Pi Zero W 2 and, of course, RaspAP. It worked fine almost immediately, on both microcomputers. Quite happy I've changed the admin password and the raspap-webgui password. I've properly shut down everything and gone to bed. This morning none of the two Pi works as an extender anymore. The two microcomputers have:
1. Raspbian version 11 Bullseye (32bit Lite) 
2. Kernel version 6.1.21-v7+
3. Full upgrade run as the first step
4. latest raspAp (downloaded two days ago)
5. Instructions to instal RaspAP and setup a WiFi Extender followed to the letter

The Pi Zero has an ETH/USB HUB HAT by waveshare and, therefore, in both cases I can always connect with an ethernet cable  to eth0.
In both cases wlan1 is connected to the home wifi  with IP 192.168.1.xy and wlan0 has IP address 10.3.141.1. The point is that there is no raspap-webgui on the air.
If I run ifconfig I get:  (note: ethernet cable disconnected as I can login with SSH through wlan1)
eth0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        ether b8:27:eb:d6:b3:69  txqueuelen 1000  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 84  bytes 5592 (5.4 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 84  bytes 5592 (5.4 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

wlan0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.3.141.1  netmask 255.255.255.0  broadcast 10.3.141.255
        inet6 fe80::d5d2:3e85:537b:5c23  prefixlen 64  scopeid 0x20<link>
        ether 34:7d:e4:4f:bb:51  txqueuelen 1000  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 52 overruns 0  carrier 0  collisions 0

wlan1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.1.77  netmask 255.255.255.0  broadcast 192.168.1.255
        inet6 fe80::b1ab:b502:6a6f:54e3  prefixlen 64  scopeid 0x20<link>
        ether 34:7d:e4:4f:b0:67  txqueuelen 1000  (Ethernet)
        RX packets 97  bytes 20291 (19.8 KiB)
        RX errors 0  dropped 7  overruns 0  frame 0
        TX packets 66  bytes 12571 (12.2 KiB)
        TX errors 0  dropped 2 overruns 0  carrier 0  collisions 0

I cannot restart raspap.services as it doesn't find it and even if I copy it from /var/www/html/installers to /lib/systemd/system/, releoad the daemons and enable raspapd.service nothing happens and at reboot raspapd.service has disappeared.
Has anybody an idea about what could have happened and how to solve it.
Many thank for any help

**Comment by billz** [comment]
Hmm, that's odd. I suspect it could be related to the daemon not loading. A normal raspapd runs at startup and exits (`RemainAfterExit=no`). The following is typical:

```
○ raspapd.service - RaspAP Service Daemon
     Loaded: loaded (/lib/systemd/system/raspapd.service; enabled; preset: enabled)
     Active: inactive (dead) since Sat 2024-03-09 16:14:54 CET; 2s ago
    Process: 3307 ExecStart=/bin/bash /etc/raspap/hostapd/servicestart.sh --interface uap0 --seconds 3 (code=exited, status=0/SUCCESS)
   Main PID: 3307 (code=exited, status=0/SUCCESS)
        CPU: 367ms

Mar 09 16:14:40 raspberrypi systemd[1]: Starting raspapd.service - RaspAP Service Daemon...
Mar 09 16:14:40 raspberrypi bash[3307]: Stopping network services...
Mar 09 16:14:41 raspberrypi bash[3307]: Changing RaspAP Daemon --interface to uap0
Mar 09 16:14:41 raspberrypi bash[3307]: Disabling systemd-networkd
Mar 09 16:14:44 raspberrypi bash[3307]: Starting network services...
Mar 09 16:14:54 raspberrypi bash[3307]: RaspAP service start DONE
Mar 09 16:14:54 raspberrypi systemd[1]: raspapd.service: Deactivated successfully.
Mar 09 16:14:54 raspberrypi systemd[1]: Finished raspapd.service - RaspAP Service Daemon.
```

You could also try running the ExecStart manually: `/etc/raspap/hostapd/servicestart.sh --interface uap0 --seconds 3`. If this works try adding it to cron.

That said, the raspapd.service isn't strictly necessary if hostpad, dnsmasq and dhcpcd are all enabled.

**Reply by billz** [reply]
As an addendum, you may check the state of the above services with `sudo systemctl list-unit-files --type=service`

**Reply by Fabri54** [reply]
Thank you. I've followed your suggestions but to no success. The ifconfig gives the same results I've posted.  and the command systemctl list-unit-files... gives:
```
UNIT FILE                                  STATE           VENDOR PRESET
[edited for clarity -bz]
dhcpcd.service                             enabled         enabled      
dnsmasq.service                            enabled         enabled      
hostapd.service                            enabled         enabled      
openvpn.service                            enabled         enabled             
raspapd.service                            enabled         enabled      
```
I've noticed that both the raspapd.service and hostapd.service are  enabled, likewise the dhclpcd.service and the dnsmasq.service. 
by the way, I forgot one thing about the configuration of my devices. I'm not using the onboard WiFi (disabled with dtoverlay and blacklisted as well) I'm using two external WiFi dongle AC850 by BrosTrend .

I'm lost also because I'm not an expert, on the contrary, sort of old (age included) beginner

**Reply by billz** [reply]
Relevant services are enabled, including raspapd. I rebooted a repeater on a test bench and it came back up (onboard wireless).

It's difficult to advise you further as I don't have your hardware. Best advice is to perform a self-diagnosis on each device using the Debug log generator (I can't review the logs personally, sorry). Perhaps others can chime in.

**Reply by Fabri54** [reply]
Well... many thanks in any case. 
Just one last question, how can I enable the debug log generator if I cannot access  it from the webpage? Is there a way of doing it from the terminal connected to the raspi via  SSH?

**Reply by billz** [reply]
You certainly can. From the default web root `/var/www/html` execute:

```
installers/debuglog.sh
```

to save the output to a file in your home directory:

```
installers/debuglog.sh > ~/debug.log
```

**Reply by Fabri54** [reply]
Thank you. Just done and rebooted. Let's see what's happening

**Reply by Fabri54** [reply]
OK, I've a nice debug.log file which tells a huge amount of info.  Should I look for help keeping this discussion open and hoping somebody can chime in or should I look for help somewhere else?
Again, may thanks for your help

**Reply by billz** [reply]
No worries. Best to upload the log to https://pastebin.com/ and post a link here.

**Reply by Fabri54** [reply, marked as answer]
Just a short update. I've restart from scratch the installation on the Pi 3B+
Re-installed the OS, updated, kept the on-board wifi alive, no eth0 connection. Installed the drivers for the WiFi dongle.
Installed RaspAP and configured as a WiFi Extender. Worked like a charm and has stood several reboots and shutdowns with disconnection from the power.  So far I haven't seen the smallest glitch.
Next I'm going to kill the on-board WiFi and add a second WiFi dongle. We'll see
